Jonathan/add hover card

Eggbask opened the pull request:
## Description

Please provide a brief description of the changes.

- Created `hover-card.tsx` component in the `components/ui/` directory 
- Created `HoverCard` folder and `HoverCard.stories.tsx` in `stories` directory 
- Replaced code in `test/page.tsx` with HoverCard component test

## Related Issue

Closes #17 

## Type of Change

- [ ] Bug fix
- [x] New feature
- [x] Documentation update
- [ ] Refactoring

## Checklist

- [x] run npm run lint to check for errors
- [x] run npm run prettier to fix styling

aaryanmohta04 commented:
Hello! While the demo resembles the designs very closely, we need to make sure that we are making a component library that is reusable. To do this, the hover card content should be styled by the one who uses the component, and should not have default content in it(like the codelab logo it does right now). Refer to the shadcn website to see how a sample hover card is implemented. I like the approach you chose by taking input as parameters(example, description), but to make it dynamic, we should give the user the ability to add and remove whatever they want inside it. Please make sure to make these changes and reach out to mentors if the instructions are not clear.

Eggbask commented:
I have removed the default content from the hover-card.tsx file and implemented it in the page.tsx file. The hover-card implementation now applies styling to the text in the Trigger and padding/margins/shadow to the content box, but doesn't contain anything by default.
